_drafts/2017-10-03-googlelanguager.md

```diff
@@ -0,0 +1,192 @@
+
+<span style="text-align:center">
+<div align="center"><img src="/assets/blog-images/2017-10-03-googlelanguager/alexa.jpg"></div>
+</span>
+
+*The voice activated internet connected speaker, Amazon's Alexa - image from www.amazon.co.uk*
+
+### Translate everything
+
+Finally, I live and work in Denmark.  As Danish is only spoken by less than 6 million people, applications that work in English may not be available in Danish very quickly, if at all.  The API's translation service is the one that made the news in 2016 for ["inventing its own language"](https://research.googleblog.com/2016/09/a-neural-network-for-machine.html), and offers much better English to Danish translations that the free web version and may make services available in Denmark sooner.
+
+## Using the library
+
+To use these APIs within R, you first need to do a one-time setup to create a Google Project, add a credit card and authenticate which is [detailed on the package website](http://code.markedmondson.me/googleLanguageR/#installation).
+
+After that, you feed in the R objects you want to operate upon.  The `rOpenSci` review helped to ensure that this can scale up easily, so that you can feed in large character vectors which the library will parse and rate limit as required.  The functions also work within [tidyverse](https://www.tidyverse.org/) pipe syntax.
+
+### Speech-to-text
+
+The [Cloud Speech API](http://code.markedmondson.me/googleLanguageR/articles/speech.html) is exposed via the [`gl_speech`](http://code.markedmondson.me/googleLanguageR/reference/gl_speech.html) function. 
+
+It supports multiple audio formats and languages, and you can either feed a sub-60 second audio file directly, or perform asynchrnous requests for longer audio files.
+
+Example code:
+
+```r
+library(googleLanguageR)
+
+my_audio <- "my_audio_file.wav"
+gl_speech(my_audio)
+#  A tibble: 1 x 3
+#  transcript confidence                 words
+#* <chr>      <dbl>                <list>
+#1 Hello Mum  0.9227779 <data.frame [19 x 3]>
+```
+
+### Translation
+
+The [Cloud Translation API](http://code.markedmondson.me/googleLanguageR/articles/translation.html) lets you translate text via [`gl_translate`](http://code.markedmondson.me/googleLanguageR/reference/gl_translate.html)
+
+As you are charged per character, one tip here if you are working with lots of different languages is to perform detection of language offline first using another `rOpenSci` package, [`cld2`](https://github.com/ropensci/cld2).  That way you can avoid charges for text that is already in your target language i.e. English.
```

Comment:
When referring to rOpenSci, no need for backticks - normal text is fine, and since this will be posted on our blog, no need to link to site or twitter